// next-session/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
var withSession_1 = require("./withSession");
Object.defineProperty(exports, "withSession", { enumerable: true, get: function () { return withSession_1.default; } });
var connect_1 = require("./connect");
Object.defineProperty(exports, "session", { enumerable: true, get: function () { return connect_1.default; } });
var core_1 = require("./core");
Object.defineProperty(exports, "applySession", { enumerable: true, get: function () { return core_1.applySession; } });
var compat_1 = require("./compat");
Object.defineProperty(exports, "promisifyStore", { enumerable: true, get: function () { return compat_1.promisifyStore; } });
Object.defineProperty(exports, "expressSession", { enumerable: true, get: function () { return compat_1.expressSession; } });
Object.defineProperty(exports, "Store", { enumerable: true, get: function () { return compat_1.Store; } });
var memory_1 = require("./store/memory");
Object.defineProperty(exports, "MemoryStore", { enumerable: true, get: function () { return memory_1.default; } });
__exportStar(require("./types"), exports);
